### PR TITLE
export downloadJson and downloadArraybuffer

### PR DESCRIPTION
**Pull request by SantyWang**
Re: #

### Changelog

* export downloadJson and downloadArrayBuffer for internal usage

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Review by shrinktofit — APPROVED** [review]

File: cocos/asset/asset-manager/downloader.ts [hunk 1]
```diff
@@ -262,6 +262,18 @@ export class Downloader {
      */
     public downloadScript = downloadScript;
 
+    /**
+     * @engineInternal
+     * @internal
```
Comment (minggo) [inline cocos/asset/asset-manager/downloader.ts]: Why need `@internal`?

Comment (shrinktofit) [reply cocos/asset/asset-manager/downloader.ts]: image

Comment (minggo) [reply cocos/asset/asset-manager/downloader.ts]: But only `engineInternal` does the thing. `internal` can not work.
@PPpro could you please double check it?

Comment (SantyWang) [reply cocos/asset/asset-manager/downloader.ts]: For now, we need @internal to control documentation generation.

Comment (PPpro) [reply cocos/asset/asset-manager/downloader.ts]: we've fixed the issue here https://github.com/cocos/cocos-engine/pull/15826

now we can use `engineInternal` to control both `.d.ts` and api doc generation @holycanvas

Comment (SantyWang) [reply cocos/asset/asset-manager/downloader.ts]: OK

Comment (SantyWang) [inline cocos/asset/asset-manager/downloader.ts]:
```suggestion
```

File: cocos/asset/asset-manager/downloader.ts [hunk 2]
```diff
@@ -262,6 +262,17 @@ export class Downloader {
...
     public downloadScript = downloadScript;
 
+    /**
+     * @engineInternal
+     */
+    public _downloadArrayBuffer = downloadArrayBuffer;
+
+    /**
+     * @engineInternal
+     * @internal
```
Comment (SantyWang) [inline cocos/asset/asset-manager/downloader.ts]:
```suggestion
```

**Review by minggo — APPROVED** [review]

**Review by PPpro — APPROVED** [review]

**Pull request merged**